content/0007-development-environment/rfc.md

```diff
@@ -0,0 +1,231 @@
+whether it exists or not.
+
+#### Stopping a cluster
+
+To stop a cluster, you invoke the `cluster stop` command and all the supporting
+containers will be shutdown.
+
+#### Destroying a cluster
+
+To destroy a cluster, you invoke the `cluster delete` command, which will stop
+and destroy running containers and delete any saved metadata and data
+files/directories. This can be used to start a new cluster from scratch.
+
+#### Using `kubectl`
+
+To ensure there's no confusion as to which cluster operations team members are
+accessing to perform maintenance, `kubectl` commands against the development
+cluster are invoked using `rde kubectl`, which knows how to configure kubectl
+with the proper kubeconfig file.
+
+Obviously someone can choose to populate their global kubeconfig with the
+cluster credentials and change context if they want, but any mistakes caused
+doing so is on them.
+
+#### Bootstrapping
+
+* k3d creates a 3 node Kubernetes cluster and `rde` ensures it's running and
+  healthy
+* A kubectl config file is saved the `rde` data directory.
+* Critical 3rd party resources are deployed to the cluster using `kubectl apply`.
+    * tekton
+    * knative
+* Critical 3rd party resources are deployed to the cluster using Helm.
+    * Postgresql
+    * Vault
+    * Redis
+    * Ambassador
+* Relay resources are deployed to the cluster using `kubectl apply`.
+    * relay-core CRDs
+* Helm chart configuration values are generated and populated into a
+  `values.yaml` and saved to disk.
+* Relay services are deployed to the cluster using Helm. See how container
+  image management works below.
+    * relay-core
+    * relay-api
+    * relay-ui
+
+#### Image management
+
+The key `rde` being useful is proper container image management. This is a
+feature that allows a developer to easily write code locally and run it against
+a cluster without committing and waiting for a continuous integration cycle.
+
+Container images for Relay services are initially pulled using the `latest` tag
+and retagged to `local` (e.g. `relay-api:local`) unless a `local` tag already
+exists for that image.  These local images are then imported into the cluster by
+`rde`.
+
+When a development build script for a Relay service is ran, the resulting
+docker image is tagged using `local` and if the `rde` utility exists in `$PATH`,
+a call to `rde update images` is made to instruct `rde` to import the new image
+and force a new ReplicaSet to be created for the deployment.
+
+#### Initial user
+
+`rde` has a command `rde user create`, which adds a user record to the Relay
+database.
+
+#### Persisted configuration
+
+In order to start and stop cluster, and manage deployments and users, `rde`
+accesses all the data it needs from a `config.json` file that is populated when
+the cluster is created. This contains the kubeconfig, database credentials,
+vault credentials, helm values, and references to deployment images.
```

Comment:
It could also include the initial admin record to avoid needing `rde user create` every time.

Reply:
This will be baked into the workflow that further bootstraps the relay SaaS components.